Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd
import numpy as np

In [7]:
exams = pd.DataFrame({
    'Hours':[7,4,9,12,5,3,8,13,2,8],
    'Result':[1,0,1,1,1,0,0,1,0,1]
})

In [8]:
exams

Hours  Result
0      7       1
1      4       0
2      9       1
3     12       1
4      5       1
5      3       0
6      8       0
7     13       1
8      2       0
9      8       1

In [9]:
exams.corr(method='pearson')

Hours    Result
Hours   1.000000  0.669246
Result  0.669246  1.000000

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [10]:
sleep = pd.DataFrame({
    'Sleep_hours':[7,4,9,12,5,3,8,13,2,8],
    'Satisfaction':[8,9,4,3,7,0,1,7,9,6]
})

In [12]:
sleep.corr(method='spearman')

Sleep_hours  Satisfaction
Sleep_hours      1.000000     -0.330277
Satisfaction    -0.330277      1.000000

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [13]:
health = pd.DataFrame({
    'exercise':[7,4,9,12,5,3,8,13,2,8],
    'BMI':[18,7.8,14.5,13.9,23.5,20.3,24.6,31.1,27.9,23.8]
})

In [14]:
health

exercise   BMI
0         7  18.0
1         4   7.8
2         9  14.5
3        12  13.9
4         5  23.5
5         3  20.3
6         8  24.6
7        13  31.1
8         2  27.9
9         8  23.8

In [15]:
health.corr(method='pearson')

exercise       BMI
exercise  1.000000  0.093622
BMI       0.093622  1.000000

In [16]:
health.corr(method='spearman')

exercise       BMI
exercise  1.000000  0.036474
BMI       0.036474  1.000000

CONCLUSION: If we compare both the results, we find that both Pearson (0.093622) and Spearman (0.036474) correlation indicate a weak positive relationship between "exercise" and "BMI".

Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [17]:
watching = np.random.randint(2,8,50,int)
activity = np.random.randint(1,5,50,int)

In [20]:
A = pd.DataFrame(watching,columns=['TV_watching'])
B = pd.DataFrame(activity,columns=['Activity_hours'])

In [21]:
survey = pd.concat([A,B],axis=1)

In [23]:
survey.head()

TV_watching  Activity_hours
0            3               4
1            7               3
2            5               4
3            5               3
4            3               2

In [24]:
survey.corr(method='pearson')

TV_watching  Activity_hours
TV_watching        1.000000       -0.153743
Activity_hours    -0.153743        1.000000

Q5. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [25]:
calls_per_day = np.random.randint(8,35,30,int)
sales_per_week = np.random.randint(28,62,30,int)

In [30]:
A1 = pd.DataFrame(calls_per_day,columns=['Calls/day'])
B1 = pd.DataFrame(sales_per_week,columns=['Sales/week'])

In [31]:
stats = pd.concat([A1,B1],axis=1)

In [32]:
stats.head()

Calls/day  Sales/week
0         10          52
1         34          41
2         14          34
3         20          54
4         24          38

In [34]:
stats.corr(method='pearson')

Calls/day  Sales/week
Calls/day    1.000000    0.095411
Sales/week   0.095411    1.000000